e2e/tests/nightly/genomic-extraction-to-vcf.spec.ts

```diff
@@ -0,0 +1,228 @@
+    await genomicExtractionsHistorySidebar.open();
+
+    const historyTable = genomicExtractionsHistorySidebar.getHistoryTable();
+    await historyTable.waitUntilVisible();
+
+    // Verify table column names.
+    const columnNames = await historyTable.getColumnNames();
+    expect(columnNames).toEqual(
+      expect.arrayContaining(['DATASET NAME', 'STATUS', 'DATE STARTED', 'COST', 'SIZE', 'DURATION'])
+    );
+
+    // Verify row count is 1.
+    const rowCount = await historyTable.getRowCount();
+    expect(rowCount).toEqual(1);
+
+    // Verify dataset name (Column #1 is 'Dataset Name').
+    const tableCellValue = await historyTable.getCellValue(1, 1);
+    expect(tableCellValue).toEqual(datasetName);
+
+    await genomicExtractionsHistorySidebar.close();
+
+    // >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
+    // LONG WAIT: Wait for VCF files extraction and runtime creation to finish.
+    await waitForComplete(page, maxWaitTime);
+
+    // >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
+    // RUN NOTEBOOK CODE.
+    const notebookPage = await notebookPreviewPage.openEditMode(notebookName);
+    // Run all cells. Run one cell at a time is slower.
+    await notebookPage.runAllCells();
+    await notebookPage.waitForKernelIdle(5 * 60 * 1000, 5000);
+    await notebookPage.save();
+
+    // Find any output_error.
+    const frame = await notebookPage.getIFrame();
+    const hasOutputError = (await frame.$$('.cell.code_cell .output_subarea.output_error')).length > 0 ? true : false;
+    expect(hasOutputError).toBe(false);
+
+    // Check all seven code cells have ran.
+    const prompts = await page.$$('.input .prompt_container .input_prompt');
+    expect(prompts.length).toBe(7);
+
+    // Spot-check on some cell outputs.
+    const [, cell2Output] = await notebookPage.getCellInputOutput(2);
+    expect(cell2Output).toMatch(/^VCF extraction has completed, continuing$/);
+
+    // cell #7 output is html table.
+    const cell7 = await notebookPage.findLastCell();
+    const cell7OutputElement = await cell7.findOutputElementHandle();
+    const [imgElement] = await cell7OutputElement.$x('./table');
+    expect(imgElement).toBeTruthy();
+
+    // >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
+    // Clean up.
+    await notebookPage.deleteNotebook(notebookName);
+
+    logger.info('Deleting runtime');
+    await runtimePanel.open();
+    await runtimePanel.clickButton(LinkText.DeleteEnvironment);
+    await runtimePanel.clickButton(LinkText.Delete);
+    await runtimePanel.waitUntilClose();
+    // Not waiting for delete to finish.
+  });
+
+  // Check creation status.
+  async function waitForComplete(page: Page, maxTime: number): Promise<boolean> {
+    const pollInterval = 10 * 1000;
+    let isRuntimeReady = false;
+    let isExtractionReady = false;
+    const runtimeSidebar = new RuntimePanel(page);
+    const genomicSidebar = new GenomicExtractionsSidebar(page);
+    const startTime = Date.now();
+    while (Date.now() - startTime <= maxTime) {
```

Comment:
Do you need this loop? Can you just pass `maxTime` into `genomicSidebar.waitForComplete`?

Reply:
^